lms/services/hapi.py

```diff
@@ -0,0 +1,115 @@
+"""Hypothesis API service."""
+import json
+
+import requests
+from requests import RequestException
+
+from lms.services import HAPIError
+
+
+class HypothesisAPIService:
+    """
+    Low-level API service.
+
+    A service that provides low-level methods for calling the Hypothesis API
+    with builtin error handling. This service provides HTTP verb methods like
+    ``get()`` and ``post()``. Other services can be built on top of this one
+    and provide higher-level methods like ``create_user()`` or
+    ``update_group()``.
+    """
+
+    def __init__(self, _context, request):
+        """
+        Return a new HypothesisAPIService object.
+
+        :arg _context: the Pyramid context resource
+        :arg request: the Pyramid request
+        :type request: pyramid.request.Request
+        """
+        settings = request.registry.settings
+
+        self._client_id = settings["h_client_id"]
+        self._client_secret = settings["h_client_secret"]
+        self._authority = settings["h_authority"]
+        self._base_url = settings["h_api_url"]
+
+    def delete(self, *args, **kwargs):
+        return self.request("DELETE", *args, **kwargs)
+
+    def get(self, *args, **kwargs):
+        return self.request("GET", *args, **kwargs)
+
+    def patch(self, *args, **kwargs):
+        return self.request("PATCH", *args, **kwargs)
+
+    def post(self, *args, **kwargs):
+        return self.request("POST", *args, **kwargs)
+
+    def put(self, *args, **kwargs):
+        return self.request("PUT", *args, **kwargs)
+
+    def request(
+        self, method, path, data=None, username=None, statuses=None
+    ):  # pylint:disable=too-many-arguments
+        """
```

Comment:
This is just the `post()` method from master more or less unchanged, except to add `method` as an argument.

It may be possible to remove the `disable=too-many-arguments` in the future once we have a service for retrieving `username`.